### ISSUE TITLE
Self Signed Certificate

### ISSUE THREAD

**Issue by dkoshkin** [davideberdin/kube-monitor#2]
The requests fail silently when connecting to an endpoint with a self signed cert, any way to get this working?

I'm happy to work on it, just not sure where to start.

**Comment by davideberdin** [comment]
Can you explain a bit more your case ? I mean, are you spinning up a local server where you run the kubernetes dashboard (1) or are you trying to connect to a deployed service in which you need the certificate to connect (2) ? 
For both cases, I guess you need to create a little ```express-js``` server that will handle the requests to the deployed service or to your local service. After a quick search I found out the following: your server needs to read the certificates before starting up - see [https://nodejs.org/docs/latest/api/https.html#https_https_request_options_callback](https://nodejs.org/docs/latest/api/https.html#https_https_request_options_callback) and this [https://gist.github.com/andrewjmead/6330312](https://gist.github.com/andrewjmead/6330312).

Anyway, in both cases I'm not sure what to do because I do not have experience with certificates. If my answer is correct and it's what you meant, we can then discuss about the implementation.

**Comment by davideberdin** [comment]
With the new [commit](https://github.com/davideberdin/kube-monitor/commit/754f7fafa056e284f3b970cefb979840bed210b6) I introduce the opportunity to add the certificates to connect to the Kubernetes endpoint. I hope it works for you as well now. If not, please open another ticket, thanks :)

**Comment by dkoshkin** [comment]
Just confirmed it worked!
I provided the cert/key used when setting up the `kube-apiserver`
